src/plugins/explore/public/components/tabs/action_bar/patterns_settings/patterns_settings_popover_content.tsx

```diff
@@ -29,12 +30,12 @@ const gatherOptions = (dataset?: DataView) => {
 
   // Filter out fields that are not suitable for patterns
   // For example, exclude scripted fields or meta fields
-  const filteredFields = fields.filter((field) => {
+  const filteredFields = fields.filter((field: IndexPatternField) => {
     return (
       !field.scripted &&
       !dataset.metaFields.includes(field.name) &&
       !field.subType &&
-      ['text', 'keyword'].includes(field?.esTypes?.[0] ?? '')
+      field.type === 'string'
```

Comment:
but i dont think if field is unindexed then it shouldn't be selectable. so we should consider checking if text or keyword

Reply:
I didn't know this at the time, but the ESType has no bearing on whether the PPL `patterns` command works or not. It seems to work on fields of type `keyword`, `wildcard`, `text`, etc.
Even knowing if its not aggregatable doesn't really matter, because the PPL `patterns` command would work on it regardless. 
All of this really does just get covered by checking if the field's osd type is `"string"`